### PR TITLE
first option for time_index/freq

### LINKED ISSUE #133: Add option 'first'  for freq/time_index to summary methods
Option `last`  already exists, an option `first` can be useful to e.g. check initial values without further filtering.

### PR DESCRIPTION
**Pull request by asnyv**
Resolves #133 

Took a quick look at `ecl2df` where it seems to be supported already by [resample_smry_dates](https://github.com/equinor/ecl2df/blob/c5aeeb6e8d9955d70b95a0652d9455f16cdcd2f9/ecl2df/summary.py#L135-L136), though lacking in documentation.

### CODE REVIEW

File: src/fmu/ensemble/ensemble.py [hunk 1]
```diff
@@ -965,18 +966,20 @@ def get_smry_dates(
...
+               'first' will give out the first date (minimum).
                'last' will give out the last date (maximum).
             normalize:  Whether to normalize backwards at the start
                 and forwards at the end to ensure the raw
                 date range is covered.
             start_date: str or date with first date to include.
                 Dates prior to this date will be dropped, supplied
                 start_date will always be included. Overrides
-                normalized dates. If string, use ISO-format, YYYY-MM-DD.
+                normalized dates. . Overriden if freq is 'first' or 'last'.
```
Comment (berland) [inline src/fmu/ensemble/ensemble.py:977]: dot dot

File: src/fmu/ensemble/ensemble.py [hunk 2]
```diff
@@ -741,12 +741,13 @@ def load_smry(
                 memory simultaneously
             start_date (str or date): First date to include.
                 Dates prior to this date will be dropped, supplied
-                start_date will always be included. If string, use
+                start_date will always be included. Overriden if time_index
```
Comment (berland) [inline src/fmu/ensemble/ensemble.py:744]: Overridden

File: src/fmu/ensemble/ensemble.py (hunk 1 above)
Comment (berland) [reply src/fmu/ensemble/ensemble.py:977]: Overridden

File: src/fmu/ensemble/ensemble.py [hunk 3]
```diff
@@ -965,18 +966,20 @@ def get_smry_dates(
...
             start_date: str or date with first date to include.
                 Dates prior to this date will be dropped, supplied
                 start_date will always be included. Overrides
-                normalized dates. If string, use ISO-format, YYYY-MM-DD.
+                normalized dates. . Overriden if freq is 'first' or 'last'.
+                If string, use ISO-format, YYYY-MM-DD.
             end_date: str or date with last date to be included.
                 Dates past this date will be dropped, supplied
                 end_date will always be included. Overrides
-                normalized dates. Overriden if freq is 'last'.
```
Comment (berland) [inline src/fmu/ensemble/ensemble.py:979]: Overridden

File: src/fmu/ensemble/ensemble.py [hunk 4]
```diff
@@ -1121,12 +1127,12 @@ def get_smry_stats(
                 object in memory after data has been loaded.
             start_date: str or date with first date to include.
                 Dates prior to this date will be dropped, supplied
-                start_date will always be included. If string,
-                use ISO-format, YYYY-MM-DD.
+                start_date will always be included. Overriden if time_index
```
Comment (berland) [inline src/fmu/ensemble/ensemble.py:1130]: Overridden

File: src/fmu/ensemble/ensemble.py [hunk 5]
```diff
@@ -1427,11 +1433,12 @@ def get_smry(
                 not enough memory to keep all summary files in memory.
             start_date: str or date with first date to include.
                 Dates prior to this date will be dropped, supplied
-                start_date will always be included.
+                start_date will always be included. Overriden if time_index
```
Comment (berland) [inline src/fmu/ensemble/ensemble.py:1436]: Overridden

File: src/fmu/ensemble/ensemble.py [hunk 6]
```diff
@@ -1427,11 +1433,12 @@ def get_smry(
...
+                is 'first' or 'last'.
             end_date: str or date with last date to be included.
                 Dates past this date will be dropped, supplied
                 end_date will always be included. Overriden if time_index
```
Comment (berland) [inline src/fmu/ensemble/ensemble.py:1440]: So many old typos...

File: src/fmu/ensemble/ensemble.py [hunk 7]
```diff
@@ -1427,11 +1433,12 @@ def get_smry(
...
-                is 'last'.
+                is 'first' or 'last'.
             include_restart: boolean sent to libecl for wheter restarts
```
Comment (berland) [inline src/fmu/ensemble/ensemble.py:1442]: whether while we are at it...

File: src/fmu/ensemble/realization.py [hunk 8]
```diff
@@ -1016,20 +1016,21 @@ def load_smry(
...
+               latter will return the simulated report steps (also default).
                If a list of DateTime is supplied, data will be resampled
                to these.
             column_keys: list of column key wildcards. None means everything.
             cache_eclsum: boolean for whether to keep the loaded EclSum
                 object in memory after data has been loaded.
             start_date: str or date with first date to include.
                 Dates prior to this date will be dropped, supplied
-                start_date will always be included.
+                start_date will always be included. Overriden if time_index
```
Comment (berland) [inline src/fmu/ensemble/realization.py:1028]: search for this typo globally..

File: tests/test_ensemble.py [hunk 9]
```diff
@@ -386,16 +386,21 @@ def test_ensemble_ecl():
...
     assert len(reekensemble.get_smry_dates(freq="monthly")) == 38
     assert len(reekensemble.get_smry_dates(freq="daily")) == 1098
+    assert len(reekensemble.get_smry_dates(freq="first")) == 1
     assert len(reekensemble.get_smry_dates(freq="last")) == 1
+    assert reekensemble.get_smry_dates(freq="first") == reekensemble.get_smry_dates(
+        freq="first", start_date="1900-01-01", end_date="2050-02-01"
+    )
     assert reekensemble.get_smry_dates(freq="last") == reekensemble.get_smry_dates(
-        freq="last", end_date="2050-02-01"
+        freq="last", start_date="1900-01-01", end_date="2050-02-01"
```
Comment (berland) [inline tests/test_ensemble.py:395]: was  `start_date`  necessary here?

File: src/fmu/ensemble/realization.py (hunk 8 above)
Comment (asnyv) [reply src/fmu/ensemble/realization.py:1028]: copy-paste propagates mistakes fast 😆 will fix 👍

File: tests/test_ensemble.py (hunk 9 above)
Comment (asnyv) [reply tests/test_ensemble.py:395]: just added it to confirm that it actually overrides `start_date` as well (which I added as a new comment, previously the comment was only on `end_date`). Can remove it if you want to.

File: src/fmu/ensemble/ensemble.py (hunk 1 above)
Comment (asnyv) [reply src/fmu/ensemble/ensemble.py:977]: will fix 👍

File: src/fmu/ensemble/ensemble.py (hunk 7 above)
Comment (asnyv) [reply src/fmu/ensemble/ensemble.py:1442]: you mean e.g. `Overridden whether time_index is 'first' or 'last'`? Isn't the use of whether over if best if these two options are the only options?

Comment (berland) [reply src/fmu/ensemble/ensemble.py:1442]: No, I mean 'wheter' is an independent typo :), should be 'whether'.

**Comment by asnyv** [comment]
Also resolves #135

File: tests/test_ensemble.py (hunk 9 above)
Comment (berland) [reply tests/test_ensemble.py:395]: lgtm

**Comment by berland** [comment]
I miss a test that confirms #135 is solved. Just  initialize a scratchrealization, load monthly summary, do a to_virt() and then check that get_smry(..last/first..) works on the virtual realization.

**Pull request merged**